src/index.ts

```diff
@@ -2184,6 +2294,27 @@ export type OrderBy<T extends Resource['Read'] = AnyResourceObject> =
 			$dir: OrderByDirection;
 	  });
 
+type StringPrimitiveToType<
+	T extends 'null' | 'string' | 'number' | 'boolean' | 'Date',
+> =
+	| ('null' extends T ? null : never)
+	| ('number' extends T ? number : never)
+	| ('string' extends T ? string : never)
+	| ('boolean' extends T ? boolean : never)
+	| ('Date' extends T ? Date : never);
+type StringDictionaryToDictType<
+	T extends Dictionary<
+		Array<'null' | 'string' | 'number' | 'boolean' | 'Date'>
+	>,
+> =
+	// If it exactly equals the type `{}` then that means the literal `{}` was provided and so we shouldn't
+	// expect any keys in the result.
+	Equals<T, {}> extends true
```

Comment:
```suggestion
	keyof T extends never
```
might be marginally lighter